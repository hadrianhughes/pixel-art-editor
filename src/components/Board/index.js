import React from 'react';
import PropTypes from 'prop-types';

const Board = ({ getRef }) => (
  <canvas ref={getRef} />
);

Board.propTypes = {
  getRef: PropTypes.func,
};

Board.defaultProps = {
  getRef: () => {},
};

export default Board;
