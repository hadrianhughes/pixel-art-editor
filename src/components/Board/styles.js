import styled from 'styled-components';

import { color } from '../../styles/tools';

export const Wrapper = styled.div`
  background-color: ${color('grey')};
  height: 100%;
  position: fixed;
  width: 100%;
`;

export const Canvas = styled.canvas`
  background-color: white;
  height: 64px;
  left: 50%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%) scale(7);
  width: 64px;
`;
