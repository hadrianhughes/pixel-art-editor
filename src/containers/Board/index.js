import React from 'react';

import Board from '../../components/Board';

class BoardContainer extends React.Component {
  constructor (props) {
    super(props);

    this.el = null;
  }

  render () {
    return (
      <Board getRef={e => this.el = e} />
    );
  }
}

export default BoardContainer;
