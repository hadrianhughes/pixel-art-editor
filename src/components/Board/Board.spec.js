import React from 'react';
import renderer from 'react-test-renderer';
import Board from '../Board';

describe('Board: component', () => {
  it('Matches snapshot', () => {
    const component = renderer.create(
      <Board />
    );

    expect(component.toJSON()).toMatchSnapshot();
  })
});
