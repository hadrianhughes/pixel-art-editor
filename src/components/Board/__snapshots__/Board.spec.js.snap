// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board: component Matches snapshot 1`] = `<canvas />`;
